src/firefly/plans/xafs_scan.py

```diff
@@ -151,6 +187,39 @@ def update_wavenumber_energy(self):
             else:
                 self.update_line_edit_value(line_edit, func)
 
+    def update_total_time(self):
```

Comment:
This is a pretty long ``try…except`` block. Are you really expecting exceptions from all of this? As written, this will make it hard to debug because all exceptions will pass silently.